presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -506,6 +517,34 @@ private QueryInfo createQueryInfo(Optional<Throwable> failure)
             // TODO: create query info
             return null;
         }
+
+        private static <T> void waitFor(Collection<Future<T>> futures)
```

Comment:
There is no Airlift `MoreFutures` utility methods to do this? :)

Reply:
They all operate on `ListeableFuture`. The Spark API returns just a regular Future =\